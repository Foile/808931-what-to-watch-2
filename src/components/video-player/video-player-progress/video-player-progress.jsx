import React from "react";
import PropTypes from "prop-types";

import withPlayerDuration from "../../../hocs/with-player-duration/with-player-duration";

const VideoPlayerProgress = ({progressPosition, progressRef, toggleRef, timing}) => {
  const toggleStyles = {
    left: `${progressPosition}%`
  };

  return (
    <div className="player__controls-row">
      <div className="player__time">
        <progress
          ref={progressRef}
          value={progressPosition}
          className="player__progress"
          max="100">
        </progress>
        <div
          ref={toggleRef}
          className="player__toggler"
          style={toggleStyles}
        >
          Toggler</div>
      </div>
      <div className="player__time-value">{timing}</div>
    </div>
  );
};

VideoPlayerProgress.defaultProps = {
  progressPosition: 0,
  progressRef: {},
  toggleRef: {},
  timing: ``,
};

VideoPlayerProgress.propTypes = {
  progressPosition: PropTypes.number.isRequired,
  progressRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element)
  }).isRequired,
  toggleRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element)
  }).isRequired,
  timing: PropTypes.string.isRequired,
};

export {VideoPlayerProgress};

export default withPlayerDuration(VideoPlayerProgress);
