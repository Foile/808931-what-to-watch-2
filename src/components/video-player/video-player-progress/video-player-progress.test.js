import React from 'react';
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {VideoPlayerProgress} from './video-player-progress';

it(`VideoPlayerProgress correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<BrowserRouter>
    <VideoPlayerProgress
      progressRef={React.createRef()}
      progressPosition={20}
      toggleRef={React.createRef()}
      timing={`123`}/>
  </BrowserRouter>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

