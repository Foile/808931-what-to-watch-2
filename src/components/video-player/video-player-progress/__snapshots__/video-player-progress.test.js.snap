// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerProgress correctly renders after relaunch 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={20}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "20%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    123
  </div>
</div>
`;
